### DISCUSSION TITLE
Host Parsing

### DISCUSSION THREAD

**Discussion by PorterHoskins** [pointfreeco/swift-url-routing#33] [Q&A]
Sometimes parsing the host value is needed when working with multiple server environments or distributed enterprise systems. The urls that I need to parse are as follows:

```
https://<server>.company.io/
https://<server>.stage.company.io/
https://<server>.int.company.io/
```

Parsing the server value in the host is needed to understand which server I should be pointing my url requests at. Currently I don't believe there is a way to run a parser on the host value of a `URLRequestData`. The `Host` parser requires a static string for the entire host.

Is there a way I am not thinking about to run a parser on the host without getting the `.host` off the `URLRequestData` and running it through its own parser?

**Comment by abear247** [comment]
I just ran into the same issue today, I found a workaround that works for at least my case so I thought I would share.  We needed to set different server codes depending on the host.  As you are aware, we can't parse a host similar to how we would parse a Path for example. So you kind of want to invert the setup to parse this.

I have an enum 
```
enum ServerCode: String {
    case us = "US"
    case ca = "CA"
}
```
I then added an extension to this
```
extension ServerCode {
    static let parser = OneOf {
        Parse(.case(Self.ca)) {
            Host("app-ca.example.com")
        }
        Parse(.case(Self.us)) {
            OneOf {
                Host("app.example.com")
                Host("staging.example.com")
            }
        }
    }
}
```
This is now usable like so
```
struct ExampleParser: Parser {
    var body: some Parser<URLRequestData, AppRoute> {
       Route(.case(AppRoute.example(serverCode:))) {
            ServerCode.parser
            // everything else you want to parse
       }
    }
}
```

I know this discussion was posted a long time ago, but hope this helps someone!
